abe/helper_functions/email_helpers.py

```diff
@@ -35,6 +35,12 @@ def get_msg_list(pop_items, pop_conn):
         text = "\n".join(text)
         orig_email = email.message_from_string(text)
         messages.append(orig_email)
+        # If the email has a calendar, we are going to read it and delete it.
+        #  Otherwise, we don't want to delete the messages.
+        if orig_email.is_multipart():
+            for part in orig_email.walk():
+                if part.get_content_type() == 'text/calendar':
+                    pop_conn.dele(id)
```

Comment:
Note that this calls `pop_conn.dele(id)` twice if the message contains two text/calendar attachments. Better to test whether to delete it, and then do so. This could be done by setting a variable to False outside the loop and then True inside; or via a comprehension:

```python
if origin_email.is_multipart() and \
  any(part.get_content_type() == 'text/calendar' for part in orig_email.walk()):
  pop_conn.dele(id)
```

or in a separate function that tells whether has message has a calendar attachment (can be separately tested, and you might use it elsewhere).

In general it's dicey to mutate an object while traversing it, because mutating it can muck up the state of the traversal (causing it to hit some items twice, or some not at all). That's not a problem here, where `pop_conn.dele(id)` changes the state of the connected POP server but presumably doesn't modify orig_email, so there's just the double-deletion issue.